libraries/AP_Scheduler/PerfInfo.cpp

```diff
@@ -18,7 +18,14 @@ void AP::PerfInfo::reset()
     max_time = 0;
     min_time = 0;
     long_running = 0;
-    log_dropped = DataFlash_Class::instance()->num_dropped();
+
+    DataFlash_Class *dataflash_perf = DataFlash_Class::instance();
+    if (dataflash_perf == nullptr) {
```

Comment:
this will only return nullptr if reset() is called before DataFlash_Class is initialized. Why is it called before ?  I don't see this in the code... PerfInfo is used only by AP_Scheduler, that calls it first in its init() method and for all vehicles that is the last method called on initialization.

What am I missing?

Reply:
@lucasdemarchi yes you are right, this break if you do not initialize a dataflash when use the AP_Sheduler. If you do not use a dataflash then it continue with the below default value.

This work like AP_InertialSensor_Backend dataflash logger when we try to log the gyro and accel data:

https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp#L324